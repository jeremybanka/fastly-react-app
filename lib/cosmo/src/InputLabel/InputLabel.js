/* @flow */

import * as React from "react";
import { FormattedMessage } from "react-intl";
import { Text } from "../";

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type InputLabelProps = {
  htmlFor: string,
  children: TextType,
};
const InputLabel = (props: InputLabelProps) => {
  return (
    <Text
      as="label"
      htmlFor={props.htmlFor}
      letterSpacing="xs"
      color="grays.6"
      fontSize="xs"
    >
      {props.children}
    </Text>
  );
};

export default InputLabel;
