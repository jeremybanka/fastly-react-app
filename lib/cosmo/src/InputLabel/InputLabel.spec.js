/* @flow */

import React from "react";
import { render } from "enzyme";
import InputLabel from "./InputLabel";

describe("InputLabel test suite", () => {
  const component = <InputLabel>Label</InputLabel>;
  it("renders the stored snapshot", () => {
    expect(render(component)).toMatchSnapshot();
  });
});
