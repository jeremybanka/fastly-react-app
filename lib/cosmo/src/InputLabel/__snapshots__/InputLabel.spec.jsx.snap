// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel test suite renders the stored snapshot 1`] = `
<label
  class="Text-sc-6c1xd4-0 fktwGx"
  color="grays.6"
  font-size="xs"
  letter-spacing="xs"
>
  Label
</label>
`;
